### PR TITLE
feat: add support for setting roleARN and useSDKCreds properties for S3 artifact repository

### PR DESCRIPTION
**Pull request by ygapon-mio**
Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo/templates/workflow-controller-config-map.yaml [hunk 1]
```diff
@@ -51,4 +56,4 @@ data:
 {{ toYaml .Values.controller.persistence | indent 6 }}{{- end }}
     {{- if .Values.controller.workflowDefaults }}
     workflowDefaults:
-{{ toYaml .Values.controller.workflowDefaults | indent 6 }}{{- end }}
+{{ toYaml .Values.controller.workflowDefaults | indent 6 }}{{- end }}
```
Comment (stefansedich) [inline charts/argo/templates/workflow-controller-config-map.yaml:59]: The newline should stay, I think the problem is that above you are missing an {{- end }}

File: charts/argo/templates/workflow-controller-config-map.yaml [hunk 2]
```diff
@@ -38,6 +38,11 @@ data:
         {{- end }}
         {{- if .Values.artifactRepository.s3.region }}
         region: {{ .Values.artifactRepository.s3.region }}
+        {{- if .Values.artifactRepository.s3.roleARN }}
+        roleARN: {{ .Values.artifactRepository.s3.roleARN }}
```
Comment (stefansedich) [inline charts/argo/templates/workflow-controller-config-map.yaml:42]: Missing {{- end }} here which is why you are seeing the lint error.

**Comment by stefansedich** [comment]
@ygapon-mio if you can just add back the removed EOF newline will get this one in!

File: charts/argo/templates/workflow-controller-config-map.yaml (hunk 1 above)
Comment (ygapon-mio) [reply charts/argo/templates/workflow-controller-config-map.yaml]: Oops. Fixed. Added {{-end}} and reverted commit to remove end line

**Comment by ygapon-mio** [comment]
Thank you for quick review and approval :)

**Pull request merged**